dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataverse/template/TemplateService.java

```diff
@@ -0,0 +1,84 @@
+import edu.harvard.iq.dataverse.engine.command.impl.UpdateDataverseCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.UpdateDataverseTemplateRootCommand;
+import edu.harvard.iq.dataverse.persistence.dataset.Template;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+import io.vavr.control.Try;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Stateless
+public class TemplateService {
+
+    private static final Logger logger = Logger.getLogger(TemplateService.class.getCanonicalName());
+
+    private EjbDataverseEngine engineService;
+    private DataverseRequestServiceBean dvRequestService;
+    private TemplateDao templateDao;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    @Deprecated
+    public TemplateService() {
+    }
+
+    @Inject
+    public TemplateService(EjbDataverseEngine engineService, DataverseRequestServiceBean dvRequestService,
+                           TemplateDao templateDao) {
+        this.engineService = engineService;
+        this.dvRequestService = dvRequestService;
+        this.templateDao = templateDao;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public Try<Dataverse> updateDataverse(Dataverse dataverse) {
+        return Try.of(() -> engineService.submit(new UpdateDataverseCommand(dataverse, null, null,
+                                                                            dvRequestService.getDataverseRequest(), null)));
+    }
+
+    /**
+     * Updates dataverse regarding if it is a templateRoot or not.
+     */
+    public Try<Dataverse> updateDataverseTemplate(Dataverse dataverse, boolean inheritTemplatesValue) {
+        return Try.of(() -> engineService.submit(new UpdateDataverseTemplateRootCommand(!inheritTemplatesValue, dvRequestService.getDataverseRequest(), dataverse)));
+    }
+
+    public Try<Dataverse> deleteTemplate(Dataverse dataverse, Template templateToDelete) {
+
+
+        if (dataverse.getDefaultTemplate() != null && dataverse.getDefaultTemplate().equals(templateToDelete)) {
+            dataverse.setDefaultTemplate(null);
+        }
+
+        dataverse.getTemplates().remove(templateToDelete);
+
+        return Try.of(() -> engineService.submit(new DeleteTemplateCommand(dvRequestService.getDataverseRequest(),
+                                                                           dataverse,
+                                                                           templateToDelete,
+                                                                           templateDao.findDataversesByDefaultTemplateId(templateToDelete.getId()))))
+                .onFailure(throwable -> logger.log(Level.SEVERE, throwable.getMessage(), throwable));
+    }
+
+    public Try<Template> cloneTemplate(Template templateIn, Dataverse dataverse, LocalDateTime localDateTime) {
```

Comment:
IMO current time could be obtained in this method instead of pasing it as param. If not then at least we should name param more precise